### ISSUE TITLE
Accep-Encoding decompression fails

### ISSUE THREAD

**Issue by Zeokat** [projectdiscovery/nuclei#75]
By default Nuclei adds automatically the header `Accept-Encoding: gzip` and it works good. But... If the user adds the `Accept-Encoding: gzip` header manually to the template, it stops to work.

To reproduce the bug, you only have to add that header to a template and you will see that matchers seems that stop working.

```
requests:
  - method: GET
    path:
      - "{{BaseURL}}"
    headers:
       User-Agent: "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:74.0) Gecko/20100101 Firefox/74.0"
       Referer: "{{Hostname}}"
       Accept-Language: "en-US"
       Accept-Encoding: gzip
    matchers:
      - type: word
        words:
            - "Lunar"
        part: body
```

I can confirm that adding `Accept-Encoding: deflate` works good, i can add this header and body response is decompressed correctly.

And to finish, i already expected that Nuclei doesn't support Brotli compression, thats why the header `Accept-Encoding: br` doesn't wok in websites that have brotli compression enabled. Maybe it can be added, don't know if exists a Go library to handle Brotli compression.

**Comment by Zeokat** [comment]
Hello,

Is there progress with this bug?

**Comment by bauthard** [comment]
Watch out for https://github.com/projectdiscovery/nuclei/pull/76, this should be handled in the next major update.

**Comment by Mzack9999** [comment]
Hi @Zeokat - Good catch! I think this is caused by the standard behavior of golang net/http library. The transport layer has a flag called `DisableCompression` which is false by default:

```
    // DisableCompression, if true, prevents the Transport from
    // requesting compression with an "Accept-Encoding: gzip"
    // request header when the Request contains no existing
    // Accept-Encoding value. If the Transport requests gzip on
    // its own and gets a gzipped response, it's transparently
    // decoded in the Response.Body. However, if the user
    // explicitly requested gzip it is not automatically
    // uncompressed.
```

Seems like we need to add automatic decompression in case the user manually provide an encoding.
Brotli compression doesn't appear to be so widespread, did you encounter cases where it would have been necessary such encoding?

**Comment by Zeokat** [comment]
@Mzack9999 Honestly i never saw any website forcing to use Brotli or a case that requires Brotli enabled. From my point of view, handle correctly gzip and deflate is all that we need.

Glad to see that you locate the issue :+1:

**Comment by Mzack9999** [comment]
Implemented in #76 - Closing the issue
